packages/@react-spectrum/datepicker/src/styles.css

```diff
@@ -141,7 +141,7 @@
 }
 
 .react-spectrum-Datepicker-dialog.react-spectrum-Datepicker-dialog {
-  width: auto;
+  max-width: fit-content;
```

Comment:
instead, you could try
```
contain: inline-size;
```
on the `:global(.spectrum-Calendar-helpText)` No idea why it's global...